django_prices/widgets.py

```diff
@@ -2,24 +2,42 @@
 from django.template.loader import render_to_string
 from prices import Money
 
-__all__ = ["MoneyInput"]
+__all__ = ["MoneyInput", "MoneyConstCurrencyInput"]
 
 
-class MoneyInput(forms.TextInput):
+class MoneyInput(forms.MultiWidget):
+    def __init__(self, choices, attrs=None):
+        widgets = [
+            forms.TextInput(attrs={"type": "number", "step": "any"}),
+            forms.Select(choices=choices),
+        ]
+        super(MoneyInput, self).__init__(widgets, attrs)
+
+    def decompress(self, value):
+        if value and isinstance(value, Money):
+            return [value.amount, value.currency]
+        if value and isinstance(value, (list, tuple)) and len(value) == 2:
+            return value
+        return [None, None]
+
+
+class MoneyConstCurrencyInput(forms.MultiWidget):
     template = "prices/widget.html"
-    input_type = "number"
 
-    def __init__(self, currency, *args, **kwargs):
+    def __init__(self, currency, attrs=None):
         self.currency = currency
-        super(MoneyInput, self).__init__(*args, **kwargs)
+        widgets = [forms.TextInput(attrs={"type": "number", "step": "any"})]
+        super(MoneyConstCurrencyInput, self).__init__(widgets, attrs)
 
-    def format_value(self, value):
-        if isinstance(value, Money):
-            return value.amount
-        return value
+    def decompress(self, value):
+        if value and isinstance(value, Money):
+            return [value.amount, self.currency]
+        if value and isinstance(value, (list, tuple)) and len(value) == 2:
+            return [value[0], self.currency]
+        return [None, self.currency]
```

Comment:
How much is `$None`? What value does that pair represent?

Reply:
It has no sense, you're right. I've changed it to `[None, None]`.